src/FileList/FileListHandler.cc

```diff
@@ -187,24 +188,9 @@ void FileListHandler::GetFileList(CARTA::FileListResponse& file_list, std::strin
                                 default:
                                     break;
                             }
-                        } else if (cc_file.isRegular(true)) {
-                            // Determine if FITS gz/bz, FITS, or HDF5 file
-                            if (IsCompressedFits(full_path)) { // checks magic number and extension
-                                file_type = CARTA::FileType::FITS;
-                                add_file = true;
-                            } else {
-                                auto magic_number = GetMagicNumber(full_path);
-                                if (magic_number == FITS_MAGIC_NUMBER) {
-                                    file_type = CARTA::FileType::FITS;
-                                    add_file = true;
-                                } else if (magic_number == HDF5_MAGIC_NUMBER) {
-                                    file_type = CARTA::FileType::HDF5;
-                                    add_file = true;
-                                } else if (region_list) {
-                                    // List all regular files in region list
-                                    add_file = true;
-                                }
-                            }
+                        } else if (!region_list && cc_file.isRegular(true)) {
+                            file_type = GuessImageType(full_path, filter_mode == CARTA::Content);
+                            add_file = filter_mode == CARTA::AllFiles || file_type != CARTA::UNKNOWN;
```

Comment:
Same here, I think AllFiles means skip the guessing.  They want the file list ASAP.